tests/test_brighterFatterKernel.py

```diff
@@ -0,0 +1,151 @@
+
+        self.assertFloatsAlmostEqual(results.outputBFK.ampKernels['amp 1'], expectation, atol=1e-5)
+
+    def test_quadratic(self):
+        """Test quadratic correlation solver.
+
+        This requires a different model for the variance, so cannot
+        use the one generated by setUp.  This model is not entirely
+        physical, but will ensure that accidental code changes are
+        detected.
+        """
+        config = cpPipe.BrighterFatterKernelSolveConfig()
+        config.correlationQuadraticFit = True
+        config.forceZeroSum = True
+        task = cpPipe.BrighterFatterKernelSolveTask(config=config)
+
+        ptc = ipIsr.PhotonTransferCurveDataset(ampNames=['amp 1'], ptcFitType='FULLCOVARIANCE',
+                                               covMatrixSide=3)
+        ptc.expIdMask['amp 1'] = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+        ptc.finalMeans['amp 1'] = [1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000, 10000]
+        ptc.rawMeans['amp 1'] = ptc.finalMeans['amp 1']
+        ptc.finalVars['amp 1'] = 9e-5 * np.square(np.array(ptc.finalMeans['amp 1'], dtype=float))
+        ptc.covariances['amp 1'] = []
+        for mean, variance in zip(ptc.finalMeans['amp 1'], ptc.finalVars['amp 1']):
+            residual = variance
+            covariance = [[variance, 0.5 * residual, 0.1 * residual],
+                          [0.2 * residual, 0.1 * residual, 0.05 * residual],
+                          [0.025 * residual, 0.015 * residual, 0.01 * residual]]
+            ptc.covariances['amp 1'].append(covariance)
+
+        ptc.gain['amp 1'] = 1.0
+        ptc.noise['amp 1'] = 5.0
+
+        results = task.run(ptc, ['this is a dummy exposure'], self.camera, {'detector': 1})
+        expectation = np.array([[-3.12500e-06, -6.25000e-06, 3.12500e-06, 1.25000e-05,
+                                 3.12500e-06, -6.25000e-06, -3.12500e-06],
+                                [-6.25000e-06, -2.50000e-05, 6.25000e-06, 4.37500e-05,
+                                 6.25000e-06, -2.50000e-05, -6.25000e-06],
+                                [3.12500e-06, 6.25000e-06, 1.03125e-04, 2.50000e-04,
+                                 1.03125e-04, 6.25000e-06, 3.12500e-06],
+                                [1.25000e-05, 4.37500e-05, 2.50000e-04, 8.50000e-04,
+                                 2.50000e-04, 4.37500e-05, 1.25000e-05],
+                                [3.12500e-06, 6.25000e-06, 1.03125e-04, 2.50000e-04,
+                                 1.03125e-04, 6.25000e-06, 3.12500e-06],
+                                [-6.25000e-06, -2.50000e-05, 6.25000e-06, 4.37500e-05,
+                                 6.25000e-06, -2.50000e-05, -6.25000e-06],
+                                [-3.12500e-06, -6.25000e-06, 3.12500e-06, 1.25000e-05,
+                                 3.12500e-06, -6.25000e-06, -3.12500e-06]])
+
+        self.assertFloatsAlmostEqual(results.outputBFK.ampKernels['amp 1'], expectation, atol=1e-5)
+
```

Comment:
You test 2 cases here, the "quadratic" one and the "average". Do you need to test the case where the "a" matrix is use? (these are the 3 cases to produce `preKernel` in `makeBrightterFatterKernel.py`).

Reply:
Yes, that should be a test case.